source/PlayState.hx

```diff
@@ -2142,7 +2142,7 @@ class PlayState extends MusicBeatState
 			}
 			combo = 0;
 
-			songScore -= 10;
+			songScore -= (songScore - 10) >= 0 ? 10 : songScore;
```

Comment:
Keeping the zero lower bound code separate from the score modification would be a bit simpler and easier to read.
```suggestion
			songScore -= 10;
			if (songScore < 0)
			{
				songScore = 0;
			}
```

Reply:
Tbh, the source code uses the ternary operator in many places so not really that hard to read, the thing with this is that there's a rare chance where this could still go to negatives if you `update()` function is called before that if statement is passed.

Reply:
Oh ouch, that would be a nasty concurrency bug.
This code is only ever run from within `update()` so I assume it should be safe. (Specifically, `update > keyShit > (noteCheck >) badNoteCheck > noteMiss`.)
I don't know for sure if HaxeFlixel is single-threaded (the original Flixel was written in ActionScript 3 which is single-threaded), but either way calling `update()` before the previous `update()` call completes seems like a recipe for total chaos.

Reply:
yea, this might be an issue later on if they add anything to call it multiple times, so thanks for suggestion anyway.